cmd/gardener-resource-manager/app/app.go

```diff
@@ -113,7 +114,10 @@ func NewResourceManagerCommand() *cobra.Command {
 				}
 			}()
 
-			if !targetClientOpts.Completed().WaitForCacheSync(ctx.Done()) {
+			ctxWaitForCache, cancelWaitForCache := context.WithTimeout(ctx, 5*time.Minute)
```

Comment:
Side-note, doesn't have to be tackled in this PR: we should further improve health checks, by separating liveness and readiness endpoint and report an unready state as long as the caches haven't synced initially. 
Though, this is also a general problem in our components.